test/fix-me.js

```diff
@@ -1,31 +1,130 @@
+      };
+
+      engine.find = function(paths, strings) {
+        expect(paths).to.have.members(['test/fixtures/code/src/code/test.js'])
+        done();
+      }
+
+      engine.run(config);
+    });
+
+    it('passes configured strings', function(done) {
+      var engine = new FixMe();
+      var engineConfig = {
+        config: {
+          strings: ['SUP']
+        }
+      };
+
+      engine.find = function(paths, strings) {
+        expect(strings).to.have.members(['SUP']);
+        done();
+      }
+
+      engine.run(engineConfig);
     });
   });
 
-  describe('#find(file)', function(){
-    it('finds and correctly prints TODO issues', function(done){
-      var capturedText = "",
-          unhookIntercept = intercept(function(txt) {
-            capturedText += txt;
-          });
+  describe('#find(paths, strings)', function() {
+    it('returns issues for instances of the given strings in the given paths', function(done) {
+      var buf = new StringIO();
+      var issues = [];
+      var engine = new FixMe(buf);
 
-      engine.find('test/fixtures/code/src/code/test.js', ["FIXME", "TODO", "HACK", "XXX", "BUG"]);
+      engine.find('test/fixtures/file.js', ['TODO', 'SUP'], function() {
+        buf.toString().split('\0').forEach(function(json) {
+          try {
+           issues.push(JSON.parse(json));
+          } catch(e) { }
+        });
 
-      // to capture standard output from engine, wait.
-      setTimeout(function() {
-          unhookIntercept();
+        expect(issues[0].type).to.eq('issue');
+        expect(issues[0].check_name).to.eq('TODO');
+        expect(issues[0].description).to.eq('TODO found');
+        expect(issues[0].categories).to.have.members(['Bug Risk']);
+        expect(issues[0].location.path).to.eq('test/fixtures/file.js');
+        expect(issues[0].location.lines.begin).to.eq(5);
+        expect(issues[0].location.lines.end).to.eq(5);
 
-          expect(capturedText).to.eq('{"type":"issue","check_name":"TODO","description":"TODO found","categories":["Bug Risk"],"location":{"path":"test/fixtures/code/src/code/test.js","lines":{"begin":5,"end":5}}}\0\n');
-          done();
-      }, 10);
+        expect(issues[1].type).to.eq('issue');
+        expect(issues[1].check_name).to.eq('SUP');
+        expect(issues[1].description).to.eq('SUP found');
+        expect(issues[1].categories).to.have.members(['Bug Risk']);
+        expect(issues[1].location.path).to.eq('test/fixtures/file.js');
+        expect(issues[1].location.lines.begin).to.eq(10);
+        expect(issues[1].location.lines.end).to.eq(10);
+
+        done();
+      });
+    });
+
+    it('skips binary files', function(done) {
+      var buf = new StringIO();
+      var issues = [];
+      var engine = new FixMe(buf);
+
+      engine.find('test/fixtures/binary.out', ['.*'], function() {
+        buf.toString().split('\0').forEach(function(json) {
+          try {
+            issues.push(JSON.parse(json));
+          } catch(e) { }
```

Comment:
Catching all exceptions when you expect the result to be empty seems counterintuitive (since unexpected exceptions, as opposed to expected exceptions, could indeed cause your result to be empty for reasons other than you were trying to test for). Is the catch needed?